apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerTest.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Diagnostics.Common;
+using Google.Cloud.Logging.V2;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Tests.Logging
+{
+    public class GoogleLoggerTest
+    {
+        private static readonly string _logMessage = "a log message";
+        private static readonly string _logName = "log-name";
+        private static readonly string _projectId = "pid";
+        private static readonly DateTime _dateTime = DateTime.UtcNow;
+        private static readonly Exception _exception = new Exception("some message");
+        private static readonly IClock _clock = GetClock();
+
+        /// <summary>
+        /// Creates an <see cref="IClock"/> that always returns the same <see cref="DateTime"/>.
```

Comment:
This one is definitely better with FakeClock, unless you're really testing for it checking the clock.